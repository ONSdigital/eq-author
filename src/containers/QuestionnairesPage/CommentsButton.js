import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Button from "components/Button";

import CommentsSvg from "./commentsIcon.svg";

const handleClick = () => {
  alert("Not implemented yet.");
};

const StyledSpan = styled.span`position: relative;`;

const StyledButton = styled(Button)`
  width: 18px;
  height: 18px;
  padding: 0.5em;
  background: url(${CommentsSvg}) no-repeat;
  background-size: contain;
  border: 0;
  opacity: 0.7;

  &:hover {
    opacity: 1;
    transition: opacity 150ms ease-in;
  }
`;

const UnreadCommentsNotifier = styled.span`
  border-radius: 100%;
  background: #006c97;
  width: 8px;
  height: 8px;
  position: absolute;
  top: -2px;
  right: -4px;
`;

const CommentsButton = ({ hasUnread }) =>
  <StyledSpan>
    <StyledButton title={"Click to see comments"} onClick={handleClick} />
    {hasUnread && <UnreadCommentsNotifier title={"You have unread comments"} />}
  </StyledSpan>;

CommentsButton.propTypes = {
  hasUnread: PropTypes.bool.isRequired
};

export default CommentsButton;
