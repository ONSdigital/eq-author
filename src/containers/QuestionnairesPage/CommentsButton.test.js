import React from "react";
import { shallow } from "enzyme";
import CommentsButton from "./CommentsButton";

describe("CommentsButton", () => {
  it("should render when there are no unread comments", () => {
    const wrapper = shallow(<CommentsButton hasUnread={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render when there are unread comments", () => {
    const wrapper = shallow(<CommentsButton hasUnread />);
    expect(wrapper).toMatchSnapshot();
  });
});
