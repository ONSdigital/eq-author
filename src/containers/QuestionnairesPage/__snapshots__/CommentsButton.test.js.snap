// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsButton should render when there are no unread comments 1`] = `
<CommentsButton__StyledSpan>
  <CommentsButton__StyledButton
    onClick={[Function]}
    title="Click to see comments"
  />
</CommentsButton__StyledSpan>
`;

exports[`CommentsButton should render when there are unread comments 1`] = `
<CommentsButton__StyledSpan>
  <CommentsButton__StyledButton
    onClick={[Function]}
    title="Click to see comments"
  />
  <CommentsButton__UnreadCommentsNotifier
    title="You have unread comments"
  />
</CommentsButton__StyledSpan>
`;
